tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml

```diff
@@ -183,6 +183,36 @@ jobs:
     artifactName: 'onnxruntime-android-full-aar'
     job_name_suffix: 'Full'
 
+- job: iOS_Full_xcframework
+  workspace:
+    clean: all
+  pool:
+    vmImage: 'macOS-10.15'
+  timeoutInMinutes:  180
+  steps:
+    - script: |
+        set -e -x
+        python3 tools/ci_build/github/apple/build_ios_framework.py \
+          --build_dir "$(Build.BinariesDirectory)/ios_framework" \
+          tools/ci_build/github/apple/default_full_ios_framework_build_settings.json
+        mkdir $(Build.BinariesDirectory)/artifacts
+        pushd $(Build.BinariesDirectory)/ios_framework/framework_out/
+        zip -vr $(Build.BinariesDirectory)/artifacts/onnxruntime_xcframework.zip onnxruntime.xcframework
```

Comment:
nit: is it possible to add a top level folder to the zip called something like onnxruntime-ios-xcframework? the extraction script that sets things up for insertion into the nuget package expects a directory containing the binaries with a trailing '-xyz' that it trims off.

https://github.com/microsoft/onnxruntime/blob/e83cc534d4bb8e7c76433adad010cce3cc6fc9d8/tools/ci_build/github/windows/extract_nuget_files.ps1#L39

Reply:
Can this be done in the download artifacts step? You can add whatever folder structure you want in the step

Reply:
Will address this in a follow up PR